src/main/java/com/playmonumenta/networkchat/utils/MessagingUtils.java

```diff
@@ -232,28 +230,72 @@ public static Component playerComponent(Player player) {
 			.replace("<hover:show_text:\"\"></hover>", "");
 		postPapiProcessing = legacyToMiniMessage(postPapiProcessing);
 		return SENDER_FMT_MINIMESSAGE.deserialize(postPapiProcessing,
-			TemplateResolver.templates(Template.template("team_color", colorMiniMessage),
-				Template.template("team_prefix", teamPrefix),
-				Template.template("team_displayname", teamDisplayName),
-				Template.template("team_suffix", teamSuffix),
-				Template.template("profile_message", profileMessage),
-				Template.template("item", () -> {
-					if (!CommandUtils.hasPermission(player, "networkchat.transform.item")) {
-						return Component.empty();
-					}
-
-					ItemStack item = player.getInventory().getItemInMainHand();
-					if (item.getType() != Material.AIR) {
-						return item.displayName().hoverEvent(item);
-					}
+			Placeholder.component("team_prefix", teamPrefix),
+			Placeholder.component("team_displayname", teamDisplayName),
+			Placeholder.component("team_suffix", teamSuffix),
+			Placeholder.component("profile_message", profileMessage),
+			Placeholder.component("item", () -> {
+				if (!CommandUtils.hasPermission(player, "networkchat.transform.item")) {
 					return Component.empty();
-			})));
+				}
+
+				ItemStack item = player.getInventory().getItemInMainHand();
+				if (item.getType() != Material.AIR) {
+					return item.displayName().hoverEvent(item);
+				}
+				return Component.empty();
+			}));
+	}
+
+	public static boolean containsPlayerMention(String text) {
+		if (!text.contains("@")) {
+			return false;
+		}
+
+		Trie<Boolean> allPlayers = new Trie<>();
+		for (String playerName : MonumentaRedisSyncAPI.getAllCachedPlayerNames()) {
+			allPlayers.put(playerName, true);
+		}
```

Comment:
This is a huge, huge list. 30k items. We should not be transforming this on every message with an @ I think. Maybe we should cache this? Or modify the Redis API to return this as a Trie directly? 

Can we just do this via regex instead? Like @[a-zA-Z0-9_] that's at most 16 characters? (or whatever the rules are for an MC name)